### ISSUE TITLE
A delay before the file preview

### ISSUE THREAD

**Issue by vifon** [ranger/ranger#202]
Scrolling through a list of videos etc. tends to be slow. Adding a small delay before the preview script is called would prevent ranger from generating previews for every file we scroll by. Would adding that feature be possible?

**Comment by nfnty** [comment]
Related to #151.

**Comment by nonamethanks** [comment]
I would like to add that this gets even worse if you are previewing a file currently in download. Since the file refreshes, ranger will spawn a ffmpegthumbnailer istance for every fraction of second that the terminal refreshes. I've seen all four cores of my CPU go to 100% in a couple of seconds with this, causing the whole system to freeze.

**Comment by tobanw** [comment]
My whole system was freezing up when ranger tried to preview large .gz files. As a workaround, I modified `scope.sh` to only preview archives below a certain file size (10MB in this case):
```sh
case "$extension" in
    # Archive extensions:
    a|ace|alz|arc|arj|bz|bz2|cab|cpio|deb|gz|jar|lha|lz|lzh|lzma|lzo|\
    rpm|rz|t7z|tar|tbz|tbz2|tgz|tlz|txz|tZ|tzo|war|xpi|xz|Z|zip)
        if [ $size -le 10485760 ]; then
            try als "$path" && { dump | trim; exit 0; }
            try acat "$path" && { dump | trim; exit 3; }
            try bsdtar -lf "$path" && { dump | trim; exit 0; }
        fi
        exit 1;;
    rar)
        ...
```

**Comment by Zekario** [comment]
@tobanw  I tried adding that to my `scope.sh`, but it seems to disable viewing any of those files, even if they're under 10MB. Would you happen to know why?

**Comment by tobanw** [comment]
@Zekario is `$size` defined? I have it defined in this section as follows:
```sh
# Find out something about the file:
...
size=$(stat -c%s "$path") # file size in bytes
```

**Comment by dahmc** [comment]
Is there any way to implement a small preview delay now after all these years?

**Comment by toonn** [comment]
@dahmc, no. Checking out the linked PRs and bringing one of them up to speed with the current code would help the process along.

**Comment by tbone2k-git** [comment]
Hi everyone! o) 

I got here, because I realized "ranger" is very slow when scrolling through files. It took me just some minutes of playing around with it to recognize the slowness and it indeed looks like the preview is called for every file, even if you only scroll "over" it. What the heck? o) 

I am testing all the file managers on Linux over time and I already wasted a lot of time so far.
pcmanfm, thunar, nemo, dolphin, krusader, mc, (insert a dozen others here), then terminal based.. fzf(?), nnn, ranger.. 

Spending an hour on each is enough to end up with all the hair pulled out or landing in a git repo and realize.. this is not the tool I am looking for, it is not even close to what I expect from a file manager in 2025. 

Is there something like "Directory Opus" (Windows application) for Linux? I make use of "Directory Opus" for almost 30 years now, I already used it back then, when it was only available for the Commodore Amiga. The modern Windows version of it seems to be multiple universes ahead of anything available for Linux and even the old Amiga version seems to blow a lot of current software for Linux out of the water. Please let me know if I just overlooked something..

How is this possible?
This is not a rant, just 100% confusion! o)

Thank you and have a nice day everyone! o)

**Comment by tex** [comment]
Try lf (https://github.com/gokcehan/lf/). I ended up with it and I am satisfied.

**Comment by tbone2k-git** [comment]
Thank you tex, "lf" seems to have the same weird implementation as "ranger", not waiting even a millisecond until the preview is generated. I noticed right away and found this related (and open) issue by accident: https://github.com/gokcehan/lf/issues/1824

"lf" also does not allow to scroll through the preview. Other basic features also seem to require tinkering with lots of scripts and whatnot. What a mess, it's the same basic toy as other file managers it seems. I think I'm done with it already.

Thank you nonetheless! o)

ps: You are not that "tex" I know from "Reinickendorf", are you? o)
